### PR TITLE
Clean stale locks

### PR DESCRIPTION
**Pull request by dooferlad**
This is a re-submit of http://reviews.vapour.ws/r/2854/, which was backed out. I hope I have addressed all concerns. This has had plenty of testing on Windows as well as Linux.

(Review request: http://reviews.vapour.ws/r/2908/)

### CODE REVIEW

File: fslock/fslock.go [hunk 1]
```diff
@@ -83,8 +131,94 @@ func (lock *Lock) heldFile() string {
...
+
+// isAlive checks that the PID given is alive by looking to see if it is the
+// current process's PID or, if it isn't, for a file named alive.<PID>, which
+// has been updated in the last 30 seconds.
+func (lock *Lock) isAlive(PID int) bool {
+	if PID == lock.PID {
+		return true
+	}
+	var misses uint
+	for misses < 10 {
```
Comment (bogdanteleaga) [inline fslock/fslock.go]: github.com/juju/retry?

Comment (dooferlad) [inline fslock/fslock.go]: Had a look but it isn't a great fit.

**Comment by dooferlad** [comment]
$$merge$$

**Comment by dooferlad** [comment]
$$merge$$

**Pull request merged**